packages/block-editor/src/components/block-list/block-async-mode-provider.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalAsyncModeProvider as AsyncModeProvider,
+	useSelect,
+} from '@wordpress/data';
+
+const BlockAsyncModeProvider = ( { children, clientId, isBlockInSelection } ) => {
```

Comment:
Nit: Conceptually speaking and if there's no impact on performance it would be better to avoid the `isBlockInSelection` as a prop and compute it here :).

Reply:
https://github.com/WordPress/gutenberg/pull/15537/commits/b0649e1f9b4d846a191993918967e20619a6d907 - I had it before but I assumed it could impact performance. I can try with the previous commit.

Reply:
I don't thin that benchmark agrees with this idea: 
```
Average time to load: 5724ms
Average time to DOM content load: 5219ms
Average time to type character: 187.28ms 😅 
Slowest time to type character: 319ms
Fastest time to type character: 158ms
```

Reply:
:)